EmployeeDirectory.BAL/Providers/EmployeeProvider.cs

```diff
@@ -5,80 +5,105 @@
 using EmployeeDirectory.BAL.Helper;
 using EmployeeDirectory.BAL.Extension;
 using Employee = EmployeeDirectory.DAL.Models.Employee;
+using EmployeeDirectory.DAL.Models;
 
 namespace EmployeeDirectory.BAL.Providers
 {
-    public class EmployeeProvider(IDataProvider data, IGetProperty prop) : IEmpProvider
+    public class EmployeeProvider(IGetProperty prop, IEmployeeOperations employeeOperations,IGetProjectDeptList list,IRoleProvider role) : IEmpProvider
     {
-        private readonly IDataProvider _dataOperations = data;
         private readonly IGetProperty _getProperty = prop;
         private static int employeeIndex;
-        private static string employeeId = "";
+        public static string employeeId = "";
+        public static string deptId = "";
+        private readonly IEmployeeOperations _employeeOperations=employeeOperations;
+        private readonly IGetProjectDeptList _employeeDeptList=list;
+        private readonly IRoleProvider _role = role;
 
         public DTO.Employee AddValueToEmp(Dictionary<string, string> values, string mode)
         {
-            if (mode == "Add")
+            if (mode.Equals("Add"))
```

Comment:
use string.Equals(string1, string2).

Please know the advantages and disadvantages of using "string1.Equals(string2)" and "string.Equals(string1, string2)"